### PR TITLE
Add authenticated gems and automated deployement

### PR DESCRIPTION
**Pull request by Herklos**

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -87,16 +93,74 @@ jobs:
...
+            sed -i -- 's/IMAGE_TAG=.*/IMAGE_TAG=latest/g' .env
+
+            # Stop the service
+            docker-compose stop
+
+            # Remove old containers
+            docker-compose rm -f
+
+            # Restart the service
+            docker-compose -f docker-compose.yml up -d
```
Comment (GuillaumeDSM) [inline .github/workflows/docker.yml:139]: wow great !

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -87,16 +93,74 @@ jobs:
...
+
   deploy_production:
-    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
+    if: github.event_name == 'push' && (startsWith(github.ref, 'refs/tags') || github.ref == 'refs/heads/master')
```
Comment (GuillaumeDSM) [inline .github/workflows/docker.yml:142]: no tag required to deploy on production?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -90,6 +91,7 @@ jobs:
 
       - name: Install gems
         run: |
+          bundle config --local without production
```
Comment (GuillaumeDSM) [inline .github/workflows/main.yml:94]: 👍

File: Gemfile [hunk 4]
```diff
@@ -64,6 +64,12 @@ group :test do
   gem 'rails-controller-testing'
 end
 
+group :production do
+  gem 'spree_credits', github: 'Drakkar-Software/spree_credits', branch: 'spree4'
+  gem 'spree_backtesting', github: 'Drakkar-Software/spree_backtesting', branch: 'spree4'
+  gem 'spree_tentacles', github: 'Drakkar-Software/spree_tentacles', branch: 'master'
```
Comment (GuillaumeDSM) [inline Gemfile:70]: 👍

File: Gemfile [hunk 5]
```diff
@@ -74,12 +80,10 @@ gem 'plotly-rails-js', github: 'Drakkar-Software/plotly-rails-js', branch: 'fina
 gem 'spree', github: 'Drakkar-Software/spree', branch: 'spree4'
 gem 'spree_auth_devise', github: 'Drakkar-Software/spree_auth_devise', branch: 'spree4'
 gem 'spree_gateway', github: 'Drakkar-Software/spree_gateway', branch: 'spree4'
-gem 'spree_digital', github: 'Drakkar-Software/spree_digital', branch: 'spree4'
```
Comment (GuillaumeDSM) [inline Gemfile:77]: 👍

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      - DATABASE_PORT=${DATABASE_PORT}
+      - RAILS_ENV=${RAILS_ENV}
+      - RAILS_MAX_THREADS=${RAILS_MAX_THREADS}
+      - SECRET_KEY_BASE=${SECRET_KEY_BASE}
+      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
+      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}
+      - S3_REGION=${S3_REGION}
+      - S3_BUCKET_NAME=${S3_BUCKET_NAME}
+      - S3_CANDLES_BUCKET_NAME=${S3_CANDLES_BUCKET_NAME}
+      - TENTACLE_SERVER_URL=${TENTACLE_SERVER_URL}
```
Comment (GuillaumeDSM) [inline docker-compose.yml:52]: 👍

**Review by GuillaumeDSM — COMMENTED** [review]
🎉

File: .github/workflows/docker.yml (hunk 2 above)
Comment (Herklos) [reply .github/workflows/docker.yml:142]: Oh yes my bad

**Review by GuillaumeDSM — APPROVED** [review]

**Pull request merged**